cluster/manager.go

```diff
@@ -958,9 +956,47 @@ func (c *ClusterManager) Start(
 	)
 	c.gossipVersion = types.GOSSIP_VERSION_2
 
+	dne := discovery.NodeEntry{
+		Id:            c.config.NodeId,
+		Ip:            c.selfNode.DataIp,
+		GossipVersion: types.GOSSIP_VERSION_2,
+	}
+	clusterInfo, err := c.discoverService.AddNode(dne)
```

Comment:
I can't understand the flow here - but does it happen after NodeInit is called ? 
If not, what happens if NodeInit fails ? 
I don't see cleanup code .. ?